x-pack/legacy/plugins/maps/public/connected_components/layer_addpanel/source_select/source_select.js

```diff
@@ -5,30 +5,33 @@
  */
 
 import React, { Fragment } from 'react';
-import { ALL_SOURCES } from '../../../layers/sources/all_sources';
+import { getLayerWizards } from '../../../layers/layer_wizard_registry';
 import { EuiTitle, EuiSpacer, EuiCard, EuiIcon } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
 import _ from 'lodash';
 
 export function SourceSelect({ updateSourceSelection }) {
-  const sourceCards = ALL_SOURCES.map(Source => {
-    const icon = Source.icon ? <EuiIcon type={Source.icon} size="l" /> : null;
+  const sourceCards = getLayerWizards().map(layerWizard => {
+    const icon = layerWizard.icon ? <EuiIcon type={layerWizard.icon} size="l" /> : null;
 
-    const sourceTitle = Source.title;
+    const onClick = () => {
+      updateSourceSelection({
+        layerWizardId: layerWizard.id,
```

Comment:
imho, we don't need an id for layer wizards. You could jsut pass in the actual wizard-instance and not need to additional `getLayerWizard` hop.